File: apps/ingestion/src/main/java/com/cloudsherpa/ingestion/connector/CloudCredentials.java
package com.cloudsherpa.ingestion.connector;

public class CloudCredentials {

  // AWS
  private String accessKeyId;
  private String secretAccessKey;

  // Azure
  private String tenantId;
  private String clientId;
  private String clientSecret;

  // GCP
  private String projectId;
  private String serviceAccountJson;

  public String getAccessKey() {
    return accessKeyId;
  }

  public void setAccessKey(String accessKey) {
    this.accessKeyId = accessKey;
  }

  public String getSecretKey() {
    return secretAccessKey;
  }

  public void setSecretKey(String secretKey) {
    this.secretAccessKey = secretKey;
  }

  public String getTenantId() {
    return tenantId;
  }

  public void setTenantId(String tenantId) {
    this.tenantId = tenantId;
  }

  public String getClientId() {
    return clientId;
  }

  public void setClientId(String clientId) {
    this.clientId = clientId;
  }

  public String getClientSecret() {
    return clientSecret;
  }

  public void setClientSecret(String clientSecret) {
    this.clientSecret = clientSecret;
  }

  public String getProjectId() {
    return projectId;
  }

  public void setProjectId(String projectId) {
    this.projectId = projectId;
  }

  public String getServiceAccountJson() {
    return serviceAccountJson;
  }

  public void setServiceAccountJson(String serviceAccountJson) {
    this.serviceAccountJson = serviceAccountJson;
  }
}

File: apps/dashboard/src/lib/fetch/dto/cloud-credentials.ts
export interface AwsCredentialsDto {
    accessKeyId: string;
    secretAccessKey: string;
}

export interface GcpCredentialsDto {
    projectId: string;
    serviceAccountJson: string;
}

export interface AzureCredentialsDto {
    subscriptionId: string;
    tenantId: string;
    clientId: string;
    clientSecret: string;
}

export interface CloudCredentials {
    accessKeyId?: string;
    secretAccessKey?: string;
    awsRegion?: string;
    tenantId?: string;
    clientId?: string;
    clientSecret?: string;
    projectId?: string;
    serviceAccountJson?: string;
}

export interface GcpCredentialsJson {
    type: string;
    project_id: string;
    private_key_id: string;
    private_key: string;
    client_email: string;
    client_id: string;
    auth_uri: string;
    token_uri: string;
}
